src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -130,107 +176,96 @@ public async Task<HttpResponseMessage> HandleRequestAsync(HttpRequestMessage req
                 }
 
-                i = path.IndexOf(EntitiesControllerSegment, StringComparison.OrdinalIgnoreCase);
-                if (i >= 0 && (request.Method == HttpMethod.Get || request.Method == HttpMethod.Post))
+                if (EntityRoute.TryMatch(path, routeValues))
                 {
-                    EntityId entityId;
-
-                    i += EntitiesControllerSegment.Length;
-                    nextSlash = path.IndexOf('/', i);
-
                     try
                     {
-                        if (nextSlash < 0)
+                        string entityName = (string)routeValues[EntityNameRouteParameter];
+                        string entityKey = (string)routeValues[EntityKeyRouteParameter];
+                        EntityId entityId = new EntityId(entityName, entityKey);
+                        if (request.Method == HttpMethod.Get)
                         {
-                            entityId = new EntityId(path.Substring(i), string.Empty);
+                            return await this.HandleGetEntityRequestAsync(request, entityId);
+                        }
+                        else if (request.Method == HttpMethod.Post)
+                        {
+                            return await this.HandlePostEntityOperationRequestAsync(request, entityId);
                         }
                         else
                         {
-                            entityId = new EntityId(path.Substring(i, nextSlash - i), path.Substring(nextSlash + 1));
+                            return request.CreateResponse(HttpStatusCode.NotFound);
                         }
                     }
                     catch (ArgumentException e)
                     {
                         return request.CreateErrorResponse(HttpStatusCode.BadRequest, e.Message);
                     }
-
-                    if (request.Method == HttpMethod.Get)
-                    {
-                        return await this.HandleGetEntityRequestAsync(request, entityId);
-                    }
-                    else
-                    {
-                        return await this.HandlePostEntityOperationRequestAsync(request, entityId);
-                    }
                 }
 
-                i = path.IndexOf(InstancesControllerSegment, StringComparison.OrdinalIgnoreCase);
-                if (i < 0)
+                if (InstancesRoute.TryMatch(path, routeValues))
                 {
-                    // Retrieve All Status or conditional query in case of the request URL ends e.g. /instances/
-                    if (request.Method == HttpMethod.Get
-                        && path.EndsWith(InstancesControllerSegment.TrimEnd('/'), StringComparison.OrdinalIgnoreCase))
+                    string instanceId = null;
+                    string operation = null;
+                    if (routeValues.ContainsKey(InstanceIdRouteParameter))

```

Comment:
Are you able to use `TryGetValue` instead? If available, that's always preferred compared to `Compare` + dictionary lookup.

Reply:
The reason I think I avoided it the first time around was because the `TryGetValue()` maps to an `object`, not a `string`, so it made the code more awkward since I had to explicitly cast the value anyways.

I think you are right that I can make it cleaner though.